vm/sdk/src/lib.rs

```diff
@@ -49,11 +49,20 @@ pub(crate) fn bytes_to_u32_vec<T: AsRef<[u8]>>(bytes: T) -> Vec<u32> {
 }
 
 pub trait VerifiableTemplate {
-  fn verify(&self, tx: &[u8], signature: &[u8; 64]) -> bool;
+  fn verify(&self, tx: Vec<u8>, signature: [u8; 64]) -> bool;
+}
+
+pub fn encode_verify(state: Vec<u8>, tx: &[u8], signature: &[u8; 64]) -> Vec<u8> {
```

Comment:
I don't think we need an `encode_verify`. The host needs to call verify when it has a raw tx, but there's no such thing as a "verify tx". When calling verify, the host doesn't need to decode any of the tx contents. It can just pass in the entire payload.

Reply:
It exists to let the host encode all the required information (account state, raw tx and its signature) into a single vector of bytes to be assigned to `msg.input_data` when it calls `vm.execute(.., msg)`. The use case I've been thinking when it is needed:

1. Host receives a signed transaction (`tx []byte, signature [64]byte`)
2. It loads the wallet state
3. It prepares `C.athcon_message` by assigning  `msg.input_data = athcon_encode_verify(state, tx, signature)`. Now `msg.input_data` contains `method_selector("athexp_verify") | state | tx | signature`.
4. It calls `C.athcon_execute(VM, msg)`
5. The VM executes `wallet.verify(state, tx, signature)` (state == &self)

Reply:
As discussed elsewhere, in order to avoid the unnecessary roundtrip described in point 3., especially given that the host _already_ has the encoded state, I suggest that we add another top-level data structure here:

ExtendedPayload := {account_state_blob byte[], ExecutionPayload}
ExecutionPayload := {MethodSelector, args byte[]}

Then the host can generate ExtendedPayload without needing to talk to the VM. ExtendedPayload here is the data structure that gets encoded and passed in as `msg.input_data`.